.azure/linux-stack.yml

```diff
@@ -25,13 +25,30 @@ jobs:
         YAML_FILE: stack-8.4.2.yaml
       stack-8.2.2:
         YAML_FILE: stack-8.2.2.yaml
+  variables:
+    STACK_ROOT: /home/vsts/.stack
```

Comment:
It should be `$(HOME)/.stack` but it seems it cant be used in build variables: https://github.com/MicrosoftDocs/vsts-docs/issues/4337

Reply:
Two possibilities:
If you can execute a command, something like `stack path --stack-root` should give you the root.
Otherwise, I think you can change the location of `~/.stack` which may also be a solution.

Reply:
`variables` is defined prior any other command, so its possible values are limited to [predefined variables and expressions](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables). We would have to set in in the `.bashrc` to use `stack path`.
I used the default one cause in macos write outside the home directory throwed permission errors but in theory the path could be where you want or  need and stack should honour it happily. In windows it must be in `C:\sr` for example.

Reply:
Ah, just in case, values defined in `variables` are set in all script steps, so `stack` must use the definition.

Reply:
I suggest adding a line to the bashrc-files that sets the stack-root since we have other env-variables there. The path uses the same trick: The path variable could only be overwritten, not appended when using the `variables` section on the yaml

Reply:
Now we could move the variable to bashrc but if we want to use it directly in the `CacheBeta` task, like it is in the windows script (and hopefully like we will able to use in linux and macos) the variable only can be in the `variables` block, cause we can't load the bashrc file in any task but `bash` one, and it limits somewhat its possible use.
Otoh there is an issue to make possible use the home directory as initial var (i only need the home directory to make sure i can write in the given path for macos)

Reply:
I didn't think of the issue that the task cannot load the bashrc file. Having a variable does indeed seem to be the best solution.